main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1569,6 +1574,46 @@ object Classpaths {
     val x2 = copyResources.value
     classDirectory.value :: Nil
   }
+  // This is a variant of exportProductsTask with tracking
+  private[sbt] def trackedExportedProducts(track: TrackLevel): Initialize[Task[Classpath]] = Def.task {
+    val art = (artifact in packageBin).value
+    val module = projectID.value
+    val config = configuration.value
+    for { (f, analysis) <- trackedProductsImplTask(track).value } yield APIMappings.store(analyzed(f, analysis), apiURL.value).put(artifact.key, art).put(moduleID.key, module).put(configuration.key, config)
+  }
+  private[this] def trackedProductsImplTask(track: TrackLevel): Initialize[Task[Seq[(File, Analysis)]]] =
+    Def.taskDyn {
+      val useJars = exportJars.value
+      val jar = (artifactPath in packageBin).value
```

Comment:
this is still going to kill the performance, as it invokes quite a lot of subtasks. Hence https://github.com/sbt/sbt/issues/2348

I'll do some graph timings with my patch to see what is invoked, but my suspicion is that this isn't going to make a massive difference to performance on Windows due to the large number of Tasks that will be created :cry: